File: Lista/ListaJava/ListaEstatica/Listavel.java
package Lista.ListaJava.ListaEstatica;

public interface Listavel {
    

    //Metodos Principais
    public void anexar(Object dado);
    public void inserir(int posicao, Object dado);
    public Object selecionar(int posicao);
    public Object[] selecionarTodos();
    public void atualizar(int posicao, Object dado);
    public Object apagar(int posicao);

    //Metodos Auxiliares
    public boolean estaVazia();
    public boolean estaCheia();
    public String imprimir();
 }

File: Lista/ListaKotlin/ListaEstatica/Listavel.kt
interface Listavel {
    // Metodos principais

    //Adiciona o dado na última posição
    fun anexar(dado: Any?)
    //Insere um dado em uma posição que não esteja ocupada
    fun inserir(posicao: Int, dado: Any?)
    //
    fun selecionar(posicao: Int): Any?
    //
    fun selecionarTodos(): Array<Any?>
    //Troca o valor da posicao informada pela a que foi passada
    fun atualizar(posicao: Int, dado: Any?)
    fun apagar(posicao: Int): Any?

    // Metodos auxiliares
    //Identico da fila os três metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}